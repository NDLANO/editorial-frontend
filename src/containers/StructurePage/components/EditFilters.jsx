/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'ndla-ui';
import { injectT } from 'ndla-i18n';
import { Plus } from 'ndla-icons/action';
import handleError from '../../../util/handleError';
import InlineEditField from './InlineEditField';
import {
  createSubjectFilter,
  editSubjectFilter,
  deleteFilter,
} from '../../../modules/taxonomy';
import WarningModal from '../../../components/WarningModal';
import EditFilterList from './EditFilterList';

class EditFilters extends React.Component {
  constructor() {
    super();
    this.state = {
      editMode: '',
    };

    this.onCancel = this.onCancel.bind(this);
    this.addFilter = this.addFilter.bind(this);
    this.showDeleteWarning = this.showDeleteWarning.bind(this);
    this.deleteFilter = this.deleteFilter.bind(this);
    this.editFilter = this.editFilter.bind(this);
  }

  onCancel() {
    this.setState({ showDelete: '' });
  }

  async addFilter(name) {
    try {
      await createSubjectFilter(this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  async editFilter(id, name) {
    try {
      await editSubjectFilter(id, this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  showDeleteWarning(filterId) {
    this.setState({ showDelete: filterId });
  }

  async deleteFilter() {
    try {
      await deleteFilter(this.state.showDelete);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
    this.setState({ showDelete: false });
  }

  render() {
    const { classes, t, filters } = this.props;
    const { editMode, showDelete } = this.state;

    return (
      <div {...classes('editFilters')} data-testid="editFilterBox">
        <EditFilterList
          filters={filters}
          editMode={editMode}
          classes={classes}
          setEditState={name => this.setState({ editMode: name })}
          showDeleteWarning={this.showDeleteWarning}
          editFilter={this.editFilter}
        />
        {editMode === 'addFilter' ? (
          <InlineEditField
            classes={classes}
            currentVal=""
            messages={{ errorMessage: t('taxonomy.errorMessage') }}
            dataTestid="addFilterInput"
            onClose={() => this.setState({ editMode: '' })}
            onSubmit={this.addFilter}
          />
        ) : (
          <Button
            stripped
            {...classes('addFilter')}
            data-testid="addFilterButton"
            onClick={() => this.setState({ editMode: 'addFilter' })}>
            <Plus />
            {t('taxonomy.addFilter')}
          </Button>
        )}
        <div {...classes('errorMessage')}>{this.state.error}</div>
        {showDelete && (
          <WarningModal
            text={t('taxonomy.confirmDelete')}
            firstAction={{ text: t('form.abort'), action: this.onCancel }}
            secondAction={{
              text: t('warningModal.delete'),
              action: this.deleteFilter,
            }}
            onCancel={this.onCancel}
          />
        )}
      </div>
    );
  }
}

EditFilters.propTypes = {
  id: PropTypes.string,
  t: PropTypes.func,
  classes: PropTypes.func,
  filters: PropTypes.arrayOf(PropTypes.object),
  getFilters: PropTypes.func,
};

export default injectT(EditFilters);
