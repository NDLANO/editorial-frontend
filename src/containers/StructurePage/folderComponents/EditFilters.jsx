/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import Button from '@ndla/button';
import { injectT } from '@ndla/i18n';
import { Plus } from '@ndla/icons/action';
import handleError from '../../../util/handleError';
import MenuItemEditField from './menuOptions/MenuItemEditField';
import {
  createSubjectFilter,
  editSubjectFilter,
  deleteFilter,
} from '../../../modules/taxonomy';
import WarningModal from '../../../components/WarningModal';
import EditFilterList from './EditFilterList';

class EditFilters extends React.Component {
  constructor() {
    super();
    this.state = {
      editMode: '',
    };
    this.addFilter = this.addFilter.bind(this);
    this.showDeleteWarning = this.showDeleteWarning.bind(this);
    this.deleteFilter = this.deleteFilter.bind(this);
    this.editFilter = this.editFilter.bind(this);
    this.setEditMode = this.setEditMode.bind(this);
  }

  setEditMode(name) {
    this.setState({ editMode: name });
  }

  async editFilter(id, name) {
    try {
      await editSubjectFilter(id, this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  showDeleteWarning(filterId) {
    this.setState({ showDelete: filterId });
  }

  async deleteFilter() {
    try {
      await deleteFilter(this.state.showDelete);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
    this.setState({ showDelete: false });
  }

  async addFilter(name) {
    try {
      await createSubjectFilter(this.props.id, name);
      this.props.getFilters();
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
    }
  }

  render() {
    const { classes, t, filters } = this.props;
    const { editMode, showDelete, error } = this.state;

    return (
      <div {...classes('editFilters')} data-testid="editFilterBox">
        <EditFilterList
          filters={filters}
          editMode={editMode}
          classes={classes}
          setEditState={this.setEditMode}
          showDeleteWarning={this.showDeleteWarning}
          editFilter={this.editFilter}
        />
        {editMode === 'addFilter' ? (
          <MenuItemEditField
            classes={classes}
            currentVal=""
            messages={{ errorMessage: t('taxonomy.errorMessage') }}
            dataTestid="addFilterInput"
            onClose={this.setEditMode}
            onSubmit={this.addFilter}
          />
        ) : (
          <Button
            stripped
            {...classes('addFilter')}
            data-testid="addFilterButton"
            onClick={() => this.setState({ editMode: 'addFilter' })}>
            <Plus />
            {t('taxonomy.addFilter')}
          </Button>
        )}
        <div {...classes('errorMessage')}>{error}</div>

        <WarningModal
          show={showDelete}
          actions={[
            {
              text: t('form.abort'),
              onClick: () => this.showDeleteWarning(),
            },
            {
              text: t('warningModal.delete'),
              'data-testid': 'warningModalConfirm',
              action: this.deleteFilter,
            },
          ]}
          text={t('taxonomy.confirmDelete')}
          onCancel={this.showDeleteWarning}
        />
      </div>
    );
  }
}

EditFilters.propTypes = {
  id: PropTypes.string,
  t: PropTypes.func,
  classes: PropTypes.func,
  filters: PropTypes.arrayOf(PropTypes.object),
  getFilters: PropTypes.func,
};

export default injectT(EditFilters);
