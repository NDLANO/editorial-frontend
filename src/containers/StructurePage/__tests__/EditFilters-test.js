/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import nock from 'nock';
import { render, Simulate, wait } from 'react-testing-library';
import EditFilters from '../components/EditFilters';

const filterMock = [
  { id: 'urn:filter:f85f8f24-9e00-4267-82f5-ffd0dd3c53fa', name: 'SF VG1' },
  { id: 'urn:filter:0a0dc7e0-911f-4674-801a-580b3e8fa09a', name: 'SF VG2' },
  {
    id: 'urn:filter:e8aa33b2-f5b7-4dee-84f6-4b594127b4c5',
    name: 'Mediesamfunnet',
  },
  {
    id: 'urn:filter:3ae5a086-8444-43c3-8fa4-db869e7292d2',
    name: 'Medieuttrykk',
  },
  { id: 'urn:filter:9e522d29-edf0-4949-bb94-a2089c79e437', name: 'SF VG3' },
];

beforeEach(() => {
  nock('http://ndla-api')
    .get('/taxonomy/v1/subjects/test/filters')
    .reply(200, filterMock);
});
const wrapper = () =>
  render(<EditFilters id="test" classes={() => {}} t={() => 'Errormelding'} />);

it('maps out filters', async () => {
  const { container } = wrapper();
  await wait();
  await wait();
  expect(container.firstChild).toMatchSnapshot();
});

it('calls add filter', async () => {
  nock('http://ndla-api')
    .post(
      '/taxonomy/v1/filters',
      JSON.stringify({ subjectId: 'test', name: 'Nytt filter' }),
    )
    .reply(201);
  nock('http://ndla-api')
    .get('/taxonomy/v1/subjects/test/filters')
    .reply(200, [...filterMock, { name: 'Nytt filter', id: 'test' }]);
  const { getByTestId, container } = wrapper();
  Simulate.click(getByTestId('addFilterButton'));

  const input = getByTestId('addFilterInput');
  input.value = 'Nytt filter';
  Simulate.change(input);
  Simulate.keyDown(input, { key: 'Enter', keyCode: 13, which: 13 });
  await wait();
  await wait();
  expect(container.firstChild).toMatchSnapshot();
});
