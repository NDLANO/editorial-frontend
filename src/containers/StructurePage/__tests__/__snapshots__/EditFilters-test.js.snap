// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls add filter 1`] = `
<div>
  <div>
    SF VG1
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:f85f8f24-9e00-4267-82f5-ffd0dd3c53fa"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:f85f8f24-9e00-4267-82f5-ffd0dd3c53fa"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    SF VG2
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:0a0dc7e0-911f-4674-801a-580b3e8fa09a"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:0a0dc7e0-911f-4674-801a-580b3e8fa09a"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    Mediesamfunnet
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:e8aa33b2-f5b7-4dee-84f6-4b594127b4c5"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:e8aa33b2-f5b7-4dee-84f6-4b594127b4c5"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    Medieuttrykk
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:3ae5a086-8444-43c3-8fa4-db869e7292d2"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:3ae5a086-8444-43c3-8fa4-db869e7292d2"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    SF VG3
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:9e522d29-edf0-4949-bb94-a2089c79e437"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:9e522d29-edf0-4949-bb94-a2089c79e437"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <button
    class="c-button c-button--stripped"
    data-testid="addFilterButton"
    type="button"
  >
    <svg
      fill="currentColor"
      preserveAspectRatio="xMidYMid meet"
      height="1em"
      width="1em"
      class="c-icon"
      viewBox="0 0 24 24"
      data-license="Apache License 2.0"
      data-source="Material Design"
      style="vertical-align: middle;"
    >
      <g>
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </g>
    </svg>
    Errormelding
  </button>
  <div />
</div>
`;

exports[`maps out filters 1`] = `
<div>
  <div>
    SF VG1
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:f85f8f24-9e00-4267-82f5-ffd0dd3c53fa"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:f85f8f24-9e00-4267-82f5-ffd0dd3c53fa"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    SF VG2
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:0a0dc7e0-911f-4674-801a-580b3e8fa09a"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:0a0dc7e0-911f-4674-801a-580b3e8fa09a"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    Mediesamfunnet
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:e8aa33b2-f5b7-4dee-84f6-4b594127b4c5"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:e8aa33b2-f5b7-4dee-84f6-4b594127b4c5"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    Medieuttrykk
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:3ae5a086-8444-43c3-8fa4-db869e7292d2"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:3ae5a086-8444-43c3-8fa4-db869e7292d2"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div>
    SF VG3
    <div
      style="display: flex;"
    >
      <button
        class="c-button c-button--stripped"
        data-testid="editFilterurn:filter:9e522d29-edf0-4949-bb94-a2089c79e437"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
      <button
        class="c-button c-button--stripped"
        data-testid="deleteFilterurn:filter:9e522d29-edf0-4949-bb94-a2089c79e437"
        type="button"
      >
        <div
          class="c-roundIcon c-roundIcon--small"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </div>
      </button>
    </div>
  </div>
  <button
    class="c-button c-button--stripped"
    data-testid="addFilterButton"
    type="button"
  >
    <svg
      fill="currentColor"
      preserveAspectRatio="xMidYMid meet"
      height="1em"
      width="1em"
      class="c-icon"
      viewBox="0 0 24 24"
      data-license="Apache License 2.0"
      data-source="Material Design"
      style="vertical-align: middle;"
    >
      <g>
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </g>
    </svg>
    Errormelding
  </button>
  <div />
</div>
`;
